File: src/main/scala/server/j/routing/HandlerDef.java
package server.j.routing;

import scala.collection.Seq;
import server.libs.Scala;

import java.util.List;

public abstract class HandlerDef {
    public abstract ClassLoader classLoader();

    public abstract String routerPackage();

    public abstract String controller();

    public abstract String method();

    protected abstract Seq<Class<?>> parameterTypes();

    public abstract String verb();

    public abstract String path();

    public abstract String comments();

    protected abstract Seq<String> modifiers();

    public List<Class<?>> getParameterTypes() {
        return Scala.asJava(parameterTypes());
    }

    public List<String> getModifiers() {
        return Scala.asJava(modifiers());
    }
}

File: src/main/scala/server/routing/HandlerDef.scala
package server.routing

/**
  * Information about a `Handler`, especially useful for loading the handler
  * with reflection.
  */
case class HandlerDef(
                       classLoader: ClassLoader,
                       routerPackage: String,
                       controller: String,
                       method: String,
                       parameterTypes: Seq[Class[_]],
                       verb: String,
                       path: String,
                       comments: String = "",
                       modifiers: Seq[String] = Seq.empty
                     ) extends server.j.routing.HandlerDef
